site-src/guides/migrating-from-ingress.md

```diff
@@ -0,0 +1,431 @@
+  annotations:
+    some-ingress-controller.example.org/tls-redirect: "True"
+spec:
+  ingressClassName: prod
+  tls:
+  - hosts:
+    - foo.example.com
+    - bar.example.com
+    secretName: example-com
+  rules:
+  - host: foo.example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: foo-app
+            port:
+              number: 80
+      - path: /orders
+        pathType: Prefix
+        backend:
+          service:
+            name: foo-orders-app
+            port:
+              number: 80
+  - host: bar.example.com
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: bar-app
+            port:
+              number: 80
+```
+
+The next three sections convert that Ingress into Gateway API resources.
+
+### Conversion Step 1 - Define Gateway
+
+The following Gateway resource:
+
+* Belongs to our GatewayClass `prod`.
+* Provisions load balancing infrastructure (this depends on the Gateway
+  implementation).
+* Configures HTTP and HTTPS listeners (entry points), which the Ingress resource
+  included implicitly:
+    * The HTTP listener `http` on port `80`
+    * The HTTPS listener `https` on port `443` with TLS termination with the
+      cert and key stored in the `example-com` Secret, which is the same Secret
+      used in the Ingress
+
+Also, note that both listeners allow all HTTPRoutes from the same namespace
+(which is the default setting) and restrict HTTPRoute hostnames to
+the `example.com` subdomain (allow hostnames like `foo.example.com` but
+not `foo.kubernetes.io`).
+
+```yaml
+{% include 'standard/simple-http-https/gateway.yaml' %}
+```
+
+### Conversion Step 2 - Define HTTPRoutes
+
+The Ingress is split into two HTTPRoutes -- one for `foo.example.com` and one
+for `bar.example.com` hostnames.
+
+```yaml
+{% include 'standard/simple-http-https/foo-route.yaml' %}
+```
+
+```yaml
+{% include 'standard/simple-http-https/bar-route.yaml' %}
+```
+
+Both HTTPRoutes:
+
+* Attach to the `https` listener of the Gateway resource from Step 1.
+* Define the same routing rules as in the Ingress rules for the corresponding
+  hostname.
+
+### Step 3 - Configure TLS Redirect
+
+The following HTTPRoute configures a TLS redirect, which the Ingress resource
+configured via an annotation. The HTTPRoute below:
+
+* Attaches to the `http` listener of our Gateway.
+* Issues a TLS redirect for any HTTP request for the `foo.example.com`
+  or `bar.example.com` hostnames.
+
+```yaml
+{% include 'standard/simple-http-https/tls-redirect-route.yaml' %}
+```
+
+## Automatic Conversion of Ingresses
+
+The [Ingress to Gateway](https://github.com/kubernetes-sigs/ingress2gateway)
+project helps translate Ingress resources to Gateway API resources, specifically
+HTTPRoutes. It is early in the development phase and still experimental in
+nature. The conversion results should always be tested and verified.
```

Comment:
I think I may have suggested this language, but in retrospect I feel like this comment will bitrot, I think it would be reasonable to drop and it and just let them follow the link to see the documentation of `ingress2gateway` which can inform them there of the current maturity and support levels:

```suggestion
HTTPRoutes. The conversion results should always be tested and verified.
```